app/src/main/java/com/github/se/signify/model/user/UserViewModel.kt

```diff
@@ -167,4 +174,53 @@ open class UserViewModel(private val repository: UserRepository) : ViewModel() {
           Log.e("UserViewModel", "Failed to remove challenge from ongoing: ${e.message}")
         })
   }
+
```

Comment:
Get the current date and check if we can unlock the next quest. Again clean code easy to review. Good job on that